File: 2015/12/rust/src/main.rs
fn main() {
    let input = std::fs::read_to_string("../input.txt").unwrap();
    // let number_regex = regex::Regex::new(r"(\d+)").unwrap();
    let number_regex = regex::Regex::new(r"(-?\d+)").unwrap();
    let total_sum = number_regex
        .find_iter(input.as_str())
        .fold(0, |acc, number| {
            return acc + number.as_str().parse::<i32>().unwrap();
        });
    println!("[Part 1] Total sum: {}", total_sum);
    // They had us on the second part, not gonna lie
    // Won't do this now.
}

File: 2015/12/python/main.py
import json

with open("input.txt", "r") as f:
    data = json.load(f)

def sum_numbers(data):
    if isinstance(data, dict):
        if "red" in data.values():
            return 0
        return sum(map(sum_numbers, data.values()))
    if isinstance(data, list):
        return sum(map(sum_numbers, data))
    if isinstance(data, int):
        return data
    return 0

print("Part 2 value:")
print(sum_numbers(data))
